enpass: init at 5.3.0

RonnyPfannschmidt opened the pull request:
###### Motivation for this change

adds the enpass password manager
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

RonnyPfannschmidt committed:
enpass: init at 5.3.0

RonnyPfannschmidt commented:
@domenkozar i think i used the wrong branch as base, where should it go

RonnyPfannschmidt commented:
oh, btw - this one competes with #18844

groxxda commented on pkgs/tools/security/enpass/default.nix:
```diff
@@ -0,0 +1,80 @@
+{stdenv, fetchurl, dpkg, patchelf, openssl, xorg
+, glib, mesa, libpulseaudio, zlib, dbus, fontconfig, freetype
+, gtk2, pango, atk, cairo, gdk_pixbuf, jasper, xkeyboardconfig
+, makeWrapper , makeDesktopItem, ...}:
```
please remove the ellipsis (...)

RonnyPfannschmidt replied:
done

groxxda commented on pkgs/tools/security/enpass/default.nix:
```diff
@@ -0,0 +1,80 @@
+, makeWrapper , makeDesktopItem, ...}:
+let
+  data = import ./data.nix;
+  uselibs = with xorg; [
```
why did you split these and didn't inline them in `libPath`?

RonnyPfannschmidt replied:
libPath is used in the wrapper script

groxxda replied:
ok sorry i somehow missed that

groxxda commented on pkgs/tools/security/enpass/default.nix:
```diff
@@ -0,0 +1,80 @@
+
+    src = fetchurl {
+      url = "http://repo.sinew.in/${data.path}";
+      sha256 = data.sha256;
```
`inherit (data) sha256;`

RonnyPfannschmidt replied:
done

groxxda commented on pkgs/tools/security/enpass/default.nix:
```diff
@@ -0,0 +1,80 @@
+
+    unpackPhase = "dpkg -X $src .";
+    installPhase=''
+    mkdir $out
```
indentation

groxxda commented on pkgs/tools/security/enpass/fet-urls.sh:
```diff
@@ -0,0 +1,6 @@
+get() {
```
what is this for?

RonnyPfannschmidt replied:
the script currently fetches the apt metadata of that repository,
i'd like to factor it into a script that returns a nix dict with the real data

the-kenny replied:
I'm voting for removing this until it's working.

RonnyPfannschmidt replied:
created a working one

groxxda commented on pkgs/tools/security/enpass/default.nix:
```diff
@@ -0,0 +1,80 @@
+let
+  data = import ./data.nix;
+  uselibs = with xorg; [
+    openssl.out
```
i believe `out` is the default. anyhow this can be removed completely because you only use it in makeLibraryPath (same for `glib`)

RonnyPfannschmidt replied:
all out uses have been undone :)

groxxda commented on pkgs/tools/security/enpass/default.nix:
```diff
@@ -0,0 +1,80 @@
+      --set QT_XKB_CONFIG_ROOT "${xkeyboardconfig}/share/X11/xkb"
+    '';
+
+
```
please consider adding a `meta` attribute

RonnyPfannschmidt replied:
suggestions?

groxxda replied:
looks ok to me now, maybe add yourself as maintainer if you want to increase the chances of getting this merged

fpletz commented:
I've changed the target branch to mster. Could you please do a rebase and squash your commits?

RonnyPfannschmidt commented:
@fpletz thanks, i also fixed a typo in the license, please have a look

RonnyPfannschmidt commented:
@fpletz for some reason the enpass icon is not added - any idea what to do about that?

RonnyPfannschmidt commented on pkgs/tools/security/enpass/default.nix:
```diff
@@ -0,0 +1,86 @@
+    };
+
+    buildInputs = [makeWrapper dpkg patchelf ];
+    phases = [ "unpackPhase" "installPhase" ];
```
i missed the correct way to do that, what documentation might help?

the-kenny replied:
Are you sure this is considered bad practice when packaging binaries? I find it acceptable here (but I might be mistaken)

the-kenny commented on pkgs/tools/security/enpass/default.nix:
```diff
@@ -0,0 +1,86 @@
+    gdk_pixbuf
+    jasper
+  ];
+    libPath = stdenv.lib.makeLibraryPath ([
```
Check your indentation - you're slipping into four spaces here where 2 spaces is common

the-kenny commented on pkgs/tools/security/enpass/default.nix:
```diff
@@ -0,0 +1,86 @@
+let
+  data = import ./data.nix;
+  # used of both wrappers and libpath
+  uselibs = with xorg; [
```
You're using `uselibs` only to create `libPath`. Can you just merge both?

RonnyPfannschmidt replied:
i ended up being able to merge

RonnyPfannschmidt commented:
@fpletz i finished updating the metadata refresh and i addressed most comments

an issue where i'd like a suggestion is mapping the current platform to the metadata
and marking the package as not building for anything else

RonnyPfannschmidt commented on pkgs/tools/security/enpass/default.nix:
```diff
@@ -0,0 +1,86 @@
+      name = "Enpass";
+      exec = "$out/bin/Enpass";
+      #icon = "Enpass";
+      comment = "password manager";
```
removed

Mic92 commented on pkgs/tools/security/enpass/update_script.py:
```diff
@@ -0,0 +1,95 @@
+from __future__ import print_function
```
Looks a little bit complicated for a simple version update script, but if it floats the boat...

RonnyPfannschmidt replied:
well, it has to load stuff from a debian repo - there is no easy install able tarball source

Mic92 commented on pkgs/tools/security/enpass/default.nix:
```diff
@@ -0,0 +1,103 @@
+, glib, mesa, libpulseaudio, zlib, dbus, fontconfig, freetype
+, gtk2, pango, atk, cairo, gdk_pixbuf, jasper, xkeyboardconfig
+, makeWrapper , makeDesktopItem, python, pythonPackages}:
+let
```
If you want to support restrict platforms, you can add this line:

`assert stdenv.system == "i686-linux" || stdenv.system == "x86_64-linux";`

RonnyPfannschmidt commented:
@Mic92 should i squash? or will it be squashed on merge?

rasendubi commented:
@RonnyPfannschmidt you better squash it yourself

RonnyPfannschmidt commented:
@domenkozar any chance of getting this backported to 16.09?

domenkozar commented:
Make a PR against 16.09 and I'll merge it
